content/docs/pipelines/metrics/output-viewer.md

```diff
@@ -37,124 +82,346 @@ output viewer. The JSON structure looks like this:
+**Metadata Fields:**
 
-## Web app
+- `source`
 
-**type:** `'web-app'`
+The `tensorboard` viewer adds a **Start Tensorboard** button to the output page. 
+
+When viewing the output page, you can:
+
+* Click **Start Tensorboard** to start a 
+  [TensorBoard](https://www.tensorflow.org/guide/summaries_and_tensorboard) Pod
+  in your Kubeflow cluster. The button text switches to **Open Tensorboard**. 
+* Click **Open Tensorboard** to open the TensorBoard interface in a new tab, 
+  pointing to the logdir data specified in the `source` field.
+
+**Note:** The Kubeflow Pipelines UI doesn't fully manage your TensorBoard 
+instances. The "Start Tensorboard" button is a convenience feature so that
+you don't have to interrupt your workflow when looking at pipeline runs. You're
+responsible for recycling or deleting the TensorBoard Pods using your Kubernetes
+management tools.
+
+**Example:**
+
+```Python
+  metadata = {
+    'outputs' : [{
+      'type': 'tensorboard',
+      'source': args.job_dir,
+    }]
+  }
+  with open('/mlpipeline-ui-metadata.json', 'w') as f:
+    json.dump(metadata, f)
+```
 
-**Metadata fields:**
+**Visualization on the Kubeflow Pipelines UI:**
 
-- `source`
+<img src="/docs/images/taxi-tip-training-step-output-tensorboard.png" 
+  alt="TensorBoard option output from a pipeline component"
+  class="mt-3 mb-3 border border-info rounded">
 
-In order to provide more flexibility rendering custom output, 
-this viewer supports specifying an HTML file that is created by the component 
-and is rendered in the outputs page as is. It's important to note that this file 
-must be self-contained, with no references to other files in the filesystem. It 
-can still have absolute references to files on the web, however. Content running 
-inside this web app is isolated in an iframe, and cannot communicate with the 
-Kubeflow Pipelines UI.
+### Web app
 
-## Markdown
-
-**type:** `'markdown'`
+**Type:** `web-app`
 
 **Metadata fields:**
 
-- `storage`
 - `source`
 
-Renders Markdown strings in the output. The Markdown data can either be read from
-a file stored remotely, or can be embedded in the metadata field `source`, in
-which case the `storage` field's value must be `'inline'`.
+The `web-app` viewer provides flexibility for rendering custom output. You can
+specify an HTML file that your component creates, and the Kubeflow Pipelines UI
+renders that HTML in the output page. The HTML file must be self-contained, with
+no references to other files in the filesystem. The HTML file can contain
+absolute references to files on the web. Content running inside the web app is
+isolated in an iframe and cannot communicate with the Kubeflow Pipelines UI.
+
+**Example:**
+
+```Python
+  static_html_path = os.path.join(output_dir, _OUTPUT_HTML_FILE)
+  file_io.write_string_to_file(static_html_path, rendered_template)
+
+  metadata = {
+    'outputs' : [{
+      'type': 'web-app',
+      'storage': 'gcs',
```

Comment:
The listed metadata fields are the required fields rather than the allowed fields. The intro stated this (line 158) but is easy to skip over. I've changed headings and added bold text to clarify.

@hongye-sun Please would you comment on whether the 'source' can be a filename in the component's local filesystem as mentioned in the above comment?

Reply:
I don't think we support local file system reference. Only gcs path is supported by pipeline UI. We have a bug to support S3 path. 

Again, @rileyjbauer, please help to confirm, thanks.

Reply:
This is actually pretty confusing, and should be refactored at some point.

In general, `source` is used to specify the path to the artifact, and can be `http[s], s3, minio, or GCS`

`storage` here is actually only used for markdown outputs where a value of `'inline'` indicates that the `source` field is markdown which can be directly rendered. Any other value for `storage` indicates that `source` is some path to the markdown file, and that path can be any of those I listed above.

Reply:
Done. PTAL.

Reply:
Oops for missing the note that the listed fields are required! Thank you for updating the headings - it's unmissable now. 😄